results/pkg/watcher/reconciler/controller.go

```diff
@@ -59,60 +59,94 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 	namespace, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
 		r.logger.Errorf("invalid resource key: %s", key)
-		return nil
+		return err
 	}
 
-	// Get the Task Run resource with this namespace/name
-	tr, err := r.pipelineclientset.TektonV1beta1().TaskRuns(namespace).Get(name, metav1.GetOptions{})
-	if errors.IsNotFound(err) {
-		// The resource no longer exists, in which case we stop processing.
-		r.logger.Infof("task run %q in work queue no longer exists", key)
-		return nil
-	} else if err != nil {
-		r.logger.Errorf("Error retrieving Result %q: %s", name, err)
+	// Get the resource object with this namespace/name
+	taskRun, terr := r.pipelineclientset.TektonV1beta1().TaskRuns(namespace).Get(name, metav1.GetOptions{})
+	pipelineRun, perr := r.pipelineclientset.TektonV1beta1().PipelineRuns(namespace).Get(name, metav1.GetOptions{})
```

Comment:
I don't think this works as expected, because the reconcile func is tied to the TaskRun informer - e.g. we only ever get notified of TaskRun changes, and not PipelineRun. This works in your test because we're normalizing the object names to `namespace/id` which we then attempt on both clients - this will break down in cases where the TaskRun name == PipelineRun name.

We should break this up into 2 funcs or structs and handle pipeline runs separately from task runs, and group similar behavior into shared funcs where appropriate - this should also reduce the various type switching we're doing.